lib/geoengineer/environment.rb

```diff
@@ -78,24 +98,23 @@ def output(id, value, &block)
 
   def all_resources
     reses = resources
-    @projects.each { |project| reses += project.all_resources }
+    @projects.values.each { |project| reses += project.all_resources }
+    @templates.values.each { |name, template| reses += template.all_resources }
     reses
   end
 
   # Factory for creating projects inside an environment
   def project(org, name, &block)
     # do not add the project a second time
-    exists = @projects.select { |p| p.org == org && p.name == name }.first
-    return exists if exists
+    return @projects[name] if @projects.key?(name)
```

Comment:
Projects should be indexed on both org and name (either combined to be org/name, or hash in hash). both infra and moonbase might have a www project, and that would cause big problems later.